components/org.wso2.carbon.identity.oauth.ciba/src/test/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaMgtDAOImplTest.java

```diff
@@ -0,0 +1,381 @@
+        }
+    }
+
+    @Test
+    public void testPersistAuthenticationSuccessStatus() throws Exception {
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, true);
+            cibaMgtDAO.updateStatus(AUTH_CODE_KEY, AuthReqStatus.AUTHENTICATED);
+        }
+
+        try (Connection connection2 = getConnection(DB_NAME)) {
+            prepareConnection(connection2, false);
+            assertEquals(cibaMgtDAO.getCibaAuthCode(AUTH_CODE_KEY).getAuthReqStatus(),
+                    AuthReqStatus.AUTHENTICATED);
+        }
+    }
+
+    @Test
+    public void testGetCibaAuthCodeKey() throws Exception {
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, false);
+            assertEquals(cibaMgtDAO.getCibaAuthCodeKey(AUTH_REQ_ID), AUTH_CODE_KEY);
+        }
+    }
+
+    @Test
+    public void testUpdateLastPollingTime() throws Exception {
+
+        long lastPolledTimeInMillis = Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC)).getTimeInMillis();
+        Timestamp lastPolledTime = new Timestamp(lastPolledTimeInMillis);
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, true);
+            cibaMgtDAO.updateLastPollingTime(AUTH_CODE_KEY, lastPolledTime);
+        }
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, false);
+            assertEquals(cibaMgtDAO.getCibaAuthCode(AUTH_CODE_KEY).getLastPolledTime(), lastPolledTime);
+        }
+    }
+
+    @Test
+    public void testUpdatePollingInterval() throws Exception {
+
+        long updatedInterval = 5;
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, true);
+            cibaMgtDAO.updatePollingInterval(AUTH_CODE_KEY, updatedInterval);
+        }
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, false);
+            assertEquals(cibaMgtDAO.getCibaAuthCode(AUTH_CODE_KEY).getInterval(), updatedInterval);
+        }
+    }
+
+
+    @Test
+    public void testGetCibaAuthCodeWithAuthReqID() throws Exception {
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, false);
+            assertEquals(cibaMgtDAO.getCibaAuthCode(AUTH_CODE_KEY).getConsumerKey(),
+                    cibaAuthCodeDO.getConsumerKey());
+        }
+    }
+
+    @Test
+    public void testGetScope() throws Exception {
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            prepareConnection(connection1, false);
+            assertEquals(cibaMgtDAO.getScope(AUTH_CODE_KEY), scopes);
+        }
+    }
+
+    protected void storeIDP() throws Exception {
+
+        String createIdp = "CREATE TABLE IF NOT EXISTS IDP (" +
```

Comment:
Add all the tables required for the tests in a .sql file and create add the start of the tests